xds/src/main/java/io/grpc/xds/XdsListenerResource.java

```diff
@@ -443,18 +443,15 @@ private static FilterChainMatch parseFilterChainMatch(
       throws ResourceInvalidException {
     ImmutableList.Builder<CidrRange> prefixRanges = ImmutableList.builder();
     ImmutableList.Builder<CidrRange> sourcePrefixRanges = ImmutableList.builder();
-    try {
-      for (io.envoyproxy.envoy.config.core.v3.CidrRange range : proto.getPrefixRangesList()) {
-        prefixRanges.add(
-            CidrRange.create(range.getAddressPrefix(), range.getPrefixLen().getValue()));
-      }
-      for (io.envoyproxy.envoy.config.core.v3.CidrRange range
-          : proto.getSourcePrefixRangesList()) {
-        sourcePrefixRanges.add(
-            CidrRange.create(range.getAddressPrefix(), range.getPrefixLen().getValue()));
-      }
-    } catch (UnknownHostException e) {
-      throw new ResourceInvalidException("Failed to create CidrRange", e);
+    for (io.envoyproxy.envoy.config.core.v3.CidrRange range : proto.getPrefixRangesList()) {
+      prefixRanges.add(
+          CidrRange.create(InetAddresses.forString(range.getAddressPrefix()),
```

Comment:
The entire point of moving the `InetAddresses.forString()` here was to make it more obvious you need to catch `IllegalArgumentException`. So catch `IllegalArgumentException`.